awspub/api.py

```diff
@@ -0,0 +1,165 @@
+    :param group: a optional group name
+    :type group: Optional[str]
+    """
+    for image_name in context.conf["images"].keys():
+        image = Image(context, image_name)
+        if group:
+            # limit the images to process to the group given on the command line
+            if group not in image.conf.get("groups", []):
+                logger.info(f"skipping image {image_name} because not part of group {group}")
+                continue
+
+        logger.info(f"processing image {image_name} (group: {group})")
+        yield image_name, image
+
+
+def create(
+    config: pathlib.Path, config_mapping: pathlib.Path, group: Optional[str]
+) -> Tuple[Dict[str, Dict[str, str]], Dict[str, Dict[str, Dict[str, str]]]]:
+    """
+    Create images in the partition of the used account based on
+    the given configuration file and the config mapping
+
+    :param config: the configuration file path
+    :type config: pathlib.Path
+    :param config_mapping: the config template mapping file path
+    :type config_mapping: pathlib.Path
+    :param group: only handles images from given group
+    :type group: Optional[str]
+    :return: the images grouped by name and by group
+    :rtype: Tuple[Dict[str, Dict[str, str]], Dict[str, Dict[str, Dict[str, str]]]
+    """
+
+    ctx = Context(config, config_mapping)
+    s3 = S3(ctx)
+    s3.upload_file(ctx.conf["source"]["path"])
+    images: List[Tuple[str, Image, Dict[str, _ImageInfo]]] = []
+    for image_name, image in _images_filtered(ctx, group):
+        image_result: Dict[str, _ImageInfo] = image.create()
+        images.append((image_name, image, image_result))
+    images_by_name, images_by_group = _images_grouped(images, group)
+    return images_by_name, images_by_group
+
+
+def list(
+    config: pathlib.Path, config_mapping: pathlib.Path, group: Optional[str]
+) -> Tuple[Dict[str, Dict[str, str]], Dict[str, Dict[str, Dict[str, str]]]]:
+    """
+    List images in the partition of the used account based on
+    the given configuration file and the config mapping
+
+    :param config: the configuration file path
+    :type config: pathlib.Path
+    :param config_mapping: the config template mapping file path
+    :type config_mapping: pathlib.Path
+    :param group: only handles images from given group
+    :type group: Optional[str]
+    :return: the images grouped by name and by group
+    :rtype: Tuple[Dict[str, Dict[str, str]], Dict[str, Dict[str, Dict[str, str]]]
+    """
+    ctx = Context(config, config_mapping)
+    images: List[Tuple[str, Image, Dict[str, _ImageInfo]]] = []
+    for image_name, image in _images_filtered(ctx, group):
+        image_result: Dict[str, _ImageInfo] = image.list()
+        images.append((image_name, image, image_result))
+
+    images_by_name, images_by_group = _images_grouped(images, group)
+    return images_by_name, images_by_group
+
+
+def public(config: pathlib.Path, config_mapping: pathlib.Path, group: Optional[str]):
```

Comment:
`make_public` would be a bit better. `public` doesn't really imply any sort of action imo

Reply:
yeah. I kind of like the single word but agree that `public` isn't good. `publish` would be another option. 
Any strong opinions?

Reply:
i'm happy with `publish`.

Reply:
for some reason (that I currently don't remember) I back then decided to not use `publish`. 
If we use `publish` now for the API, the CLI would still use `public`. Do we want that inconsistency here?

Reply:
i would change the CLI at the same time unless we can remember a compelling reason why publish didn't make sense in the cli api.

Reply:
im towards more in public since `publish` serves meaning like whole operation. For representing meaning of this function, I would use `public`.

Reply:
Ryan also said he prefers `publish`. So I added an extra commit to change the name for the API and the CLI.